nightfall-deployer/src/contract-setup.mjs

```diff
@@ -4,96 +4,62 @@ address of the contract that holds global state (State.sol)
-  const stateContractInstance = await waitForContract('State');
-  const proposerContractInstance = await waitForContract('Proposers');
-  const challengesContractInstance = await waitForContract('Challenges');
+  const tx = {
+    from: process.env.FROM_ADDRESS,
+    // value: fee,
+    gas: config.WEB3_OPTIONS.gas,
+    gasPrice: config.WEB3_OPTIONS.gasPrice,
+  };
 
-  logger.debug(
-    `transfering ownership of contracts to simpleMultiSigAddress ${simpleMultiSigAddress}`,
-  );
+  const proposers = await waitForContract('Proposers');
+  const shield = await waitForContract('Shield');
+  const challenges = await waitForContract('Challenges');
 
-  // when deploying on infura
-  // do serial registration to predict nonce
-  // or, if we have the owner's private key, sign with that, rather than use an unlocked account
-  if (config.ETH_PRIVATE_KEY) {
-    try {
-      await Web3.submitRawTransaction(
-        proposerContractInstance.methods
-          .setStateContract(stateInstance.options.address)
-          .encodeABI(),
-        proposerContractInstance.options.address,
-      );
-      await Web3.submitRawTransaction(
-        shieldContractInstance.methods.setStateContract(stateInstance.options.address).encodeABI(),
-        shieldContractInstance.options.address,
-      );
-      await Web3.submitRawTransaction(
-        challengesContractInstance.methods
-          .setStateContract(stateInstance.options.address)
-          .encodeABI(),
-        challengesContractInstance.options.address,
-      );
-    } catch (err) {
-      if (err.message.includes('Transaction has been reverted by the EVM'))
-        logger.warn(
-          'Writing contract addresses to the State contract failed. This is probably because they are aready set. Did you already run deployer?',
-        );
-      else throw new Error(err);
-    }
+  const simpleMultiSigAddress = (await waitForContract('SimpleMultiSig')).options.address;
 
-    // our last action as the deployer is to hand off our onlyOwner powers to the
-    // multisig contract
-    await Web3.submitRawTransaction(
-      shieldContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      shieldContractInstance.options.address,
-    );
-    await Web3.submitRawTransaction(
-      stateContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      stateContractInstance.options.address,
-    );
-    await Web3.submitRawTransaction(
-      proposerContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      proposerContractInstance.options.address,
-    );
-    return Web3.submitRawTransaction(
-      challengesContractInstance.methods.transferOwnership(simpleMultiSigAddress).encodeABI(),
-      challengesContractInstance.options.address,
+  const contracts = {
+    proposers,
+    shield,
+    challenges,
+  };
+
+  for await (const contractName of ['proposers', 'shield', 'challenges']) {
```

Comment:
Here better to get all the contract names by the keys of the object to avoid errors.
```
for await (const contractName of Object.keys(contracts)) {
```